homeassistant/components/glances/sensor.py

```diff
@@ -341,10 +361,32 @@ def available(self) -> bool:
         return False
 
     @property
-    def native_value(self) -> StateType:
+    def native_value(self) -> StateType | datetime:
         """Return the state of the resources."""
         value = self.coordinator.data[self.entity_description.type]
 
-        if isinstance(value.get(self._sensor_name_prefix), dict):
-            return value[self._sensor_name_prefix][self.entity_description.key]
-        return value[self.entity_description.key]
+        if isinstance(value, dict):
+            if isinstance(value.get(self._sensor_name_prefix), dict):
+                return value[self._sensor_name_prefix][self.entity_description.key]
+            return value[self.entity_description.key]
+        return value
+
+
+class GlancesTimestampSensor(GlancesSensor):
+    """Implementation of a Glances Timestamp sensor."""
+
+    @property
+    def native_value(self) -> StateType | datetime:
+        """Return the state of the resources."""
+        value = self.coordinator.data[self.entity_description.type]
+        if dt_str := re.search("\\d+:\\d+:\\d+", value):
+            dt_value = datetime.strptime(dt_str.group(), "%H:%M:%S")
+            days_re = re.match("(\\d+) day", value)
+            days = int(days_re.groups()[0]) if days_re is not None else 0
```

Comment:
Please use the util function [`parse_datetime`](https://github.com/home-assistant/core/blob/17050a328652d0aab8311afdd0e50acf4b8e3ae1/homeassistant/util/dt.py#L183) instead.

Please verify again what type you expect from the coordinator. In the code here, it looks like a string. In the test, however, you mock it to return a datetime.

Reply:
I had a quick look at `parse_datetime` and it is looking great. Thanks for the hint, I will use it to generate the datetime.

Concerning the types I think that it is right the way it is.
The api is providing a string, but the actual sensor must return a datetime object as suggested by frenck.
The test data depicts this behavior by the expected value being of type datetime and the mock data being of type string, such as a string received by the api.

Reply:
The mock is not returning a string but a datetime object. See my other comment.

Reply:
OK, I have to check it again when I have more time... I was looking at the constant `MOCK_DATA`, but jsut after a quick look I don't see any references to this at all..

Thanks for the improvements for now, I am going to work on it later!